### ISSUE TITLE
Code.string_to_quoted reporting wrong column for start of a Map

### ISSUE THREAD

**Issue by bcardarella** [elixir-lang/elixir#13591]
```elixir
iex(13)> Code.string_to_quoted!("%{a: 1}", [columns: true])
{:%{}, [line: 1, column: 2], [a: 1]}
```

I would expect the column to be `1`.

Compare this to sigils:

```elixir
iex(14)> Code.string_to_quoted!("~LVN()", [columns: true])
{:sigil_LVN, [delimiter: "(", line: 1, column: 1],
 [{:<<>>, [line: 1, column: 1], [""]}, []]}
iex(15)>
```

or even a struct:

```elixir
iex(15)> Code.string_to_quoted!("%Foo{}", [columns: true])
{:%, [line: 1, column: 1],
 [
   {:__aliases__, [line: 1, column: 2], [:Foo]},
   {:%{}, [line: 1, column: 5], []}
 ]}
```

and they each report the proper starting column.

**Comment by bcardarella** [comment]
This is in Sourceror. When matching on a map literal and patching that node the column is off by 1 character so when you go to replace that range of characters with a replacement map it ends up with `%%{a: 1}`

**Comment by bcardarella** [comment]
But if this is a `wontfix` would you suggest that in Sourceror that for just map literals the column is adjusted by `-1`?

**Comment by josevalim** [comment]
Pinging @doorgan as well then. :)

**Comment by doorgan** [comment]
We can fix this in sourceror's range function itself if it's a known issue that won't be fixed soon!

**Comment by bcardarella** [comment]
@doorgan I'll tackle that in a few minutes in the PR I have open

**Comment by josevalim** [comment]
I think the biggest issue is that, if we fix it for `map`s, you still should not offset the `%{}`inside struct. So I wonder which one is more or less work?

**Comment by lukaszsamson** [comment]
Isn’t it fixed already on master? See https://github.com/elixir-lang/elixir/commit/fd4e6b530c0e010712b06909c89820b08e49c238

**Comment by josevalim** [comment]
Thank you @lukaszsamson. I remember we had a recent discussion about this, but I didn't remember it was already fixed. :)

**Comment by bcardarella** [comment]
@lukaszsamson thank you :)

**Comment by bcardarella** [comment]
@josevalim if fixed on `master` how should Sourcer address this for the time being? I don't want to wait on the next Elixir release but I can do a patch in Sourcerer to address this temporarily and then I guess Sourceror should do a version limit up to the current minor version of Elixir then remove that patch for the next release?

**Comment by bcardarella** [comment]
I guess I could really just fix this in LiveView Native and limit on the Elixir version for now then release without the patch later

**Comment by doorgan** [comment]
@bcardarella What I used to do was to backport fixes to main to the vendored modules in Elixir. I stopped doing that when I raised the minimum elixir version requirement, but I can see us keep doing it 🤔 Or doing `Version` checks for known bugs

**Comment by bcardarella** [comment]
@josevalim I'm going to implement a fix in Sourceror using @doorgan's suggestion on branching based on version. Do you know yet what the likely version will have the fix @lukaszsamson mentioned? Next minor?

**Comment by josevalim** [comment]
1.17.0 onwards.
